### PR TITLE
[cmd/opampsupervisor] Add passthrough logging for collector

### LINKED ISSUE #35473: [cmd/opampsupervisor] Allow collector logs to passthrough to stdout
### Component(s)

_No response_

### Is your feature request related to a problem? Please describe.

When running the supervisor, collector logs are strictly directed to a file. This proves troublesome when running in a containerized environment. Collector logs can't be easily captured by the container platform's log management solution.

### Describe the solution you'd like

An option to configure the supervisor to direct collector logs to passthrough to stdout would solve this problem.

### Describe alternatives you've considered

Volume mounting the collector log file could be a valid workaround, but best practice for containers is to log to stdout by default. The user would need to worry about how to separate supervisor and collector logs, but at least this way they don't need to worry about configuring a volume mounted file.

### Additional context

This issue is somewhat related to another issue I'm working on, #35466. The solution I have for this issue works by capturing collector output and using the supervisor's logging to send them to the same place as the supervisor's. The idea being that when running in a container the supervisor would be configured to log to stdout.

### PR DESCRIPTION
**Pull request by dpaasman00**
**Description:** <Describe what has changed.>

Allow collector logs to passthrough to stdout instead of strictly being sent to a file. 

If configured to do so, the supervisor will capture collector output and log it using it's logger. This way, the supervisor should be configured to log to stdout if running in a containerized env. 

This PR follows closely with this [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35468). Right now the supervisor exclusively logs to stdout, but under the assumption that it can be configured to log elsewhere, this change uses the supervisor logger rather than setting the collector's `exec.Cmd` to log to stdout and stderr.

**Link to tracking Issue:** Closes #35473

**Testing:** <Describe what testing was performed and which tests were added.>
I built a docker image of the supervisor and collector and ran it using the `containerized` param. This is a sample of what the output looked like:

```
2024-09-30T12:42:47.472Z	DEBUG	commander/commander.go:73	Starting agent	{"agent": "/collector/observiq-otel-collector"}
2024-09-30T12:42:47.472Z	DEBUG	commander/commander.go:163	Agent process started	{"pid": 11}
2024-09-30T12:42:47.525Z	INFO	collector	commander/commander.go:156	2024-09-30T12:42:47.525Z	info	service@v0.108.1/service.go:178	Setting up own telemetry...
2024-09-30T12:42:47.525Z	INFO	collector	commander/commander.go:156	2024-09-30T12:42:47.525Z	info	service@v0.108.1/telemetry.go:98	Serving metrics	{"address": ":8888", "metrics level": "Normal"}
2024-09-30T12:42:47.525Z	INFO	collector	commander/commander.go:156	2024-09-30T12:42:47.525Z	info	service@v0.108.1/service.go:263	Starting observiq-otel-collector...	{"Version": "v2.0.0", "NumCPU": 12}
2024-09-30T12:42:47.525Z	INFO	collector	commander/commander.go:156	2024-09-30T12:42:47.525Z	info	extensions/extensions.go:38	Starting extensions...
2024-09-30T12:42:47.525Z	INFO	collector	commander/commander.go:156	2024-09-30T12:42:47.525Z	info	extensions/extensions.go:41	Extension is starting...	{"kind": "extension", "name": "opamp"}
2024-09-30T12:42:47.525Z	INFO	collector	commander/commander.go:156	2024-09-30T12:42:47.525Z	info	extensions/extensions.go:58	Extension started.	{"kind": "extension", "name": "opamp"}
2024-09-30T12:42:47.526Z	INFO	collector	commander/commander.go:156	2024-09-30T12:42:47.526Z	info	service@v0.108.1/service.go:289	Everything is ready. Begin running and processing data.
2024-09-30T12:42:47.526Z	INFO	collector	commander/commander.go:156	2024-09-30T12:42:47.526Z	info	localhostgate/featuregate.go:63	The default endpoints for all servers in components have changed to use localhost instead of 0.0.0.0. Disable the feature gate to temporarily revert to the previous default.	{"feature gate ID": "component.UseLocalHostAsDefaultHost"}
2024-09-30T12:42:47.528Z	DEBUG	commander/commander.go:220	Stopping agent process	{"pid": 11}
```

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by djaglowski — APPROVED** [review]

File: cmd/opampsupervisor/supervisor/config/config.go
Comment (evan-bradley) [inline cmd/opampsupervisor/supervisor/config/config.go:126]: Will this option change other functionality related to the Collector process? If it is only for re-emitting the Collector's logs, I think we should be more specific with the name. I see why the name makes sense having read through everything, but my first guess on seeing this is that it would affect more than just the log output.

File: cmd/opampsupervisor/supervisor/commander/commander.go
Comment (evan-bradley) [inline cmd/opampsupervisor/supervisor/commander/commander.go:1]: Could you add an E2E test (or other test if that is easier) to cover this functionality?

**Review by evan-bradley — COMMENTED** [review]
Thanks for adding this.

File: cmd/opampsupervisor/supervisor/config/config.go
Comment (dpaasman00) [reply cmd/opampsupervisor/supervisor/config/config.go:126]: No it doesn't change any other functionality related to the collector so that's a great point. I could do `passthrough_logging` to be more specific about what it does. Does that name work or is there another you have in mind?

Comment (evan-bradley) [reply cmd/opampsupervisor/supervisor/config/config.go]: That name sounds good to me.

File: cmd/opampsupervisor/supervisor/commander/commander.go
Comment (evan-bradley) [reply cmd/opampsupervisor/supervisor/commander/commander.go:1]: One idea to make this work could be to use the logging config added in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35468 to check that a known Collector log line shows up in the file the Supervisor writes.

File: cmd/opampsupervisor/supervisor/config/config.go
Comment (tigrannajaryan) [reply cmd/opampsupervisor/supervisor/config/config.go]: A few less characters to type: `passthrough_logs`?

File: cmd/opampsupervisor/supervisor/commander/commander.go
Comment (dpaasman00) [reply cmd/opampsupervisor/supervisor/commander/commander.go:1]: I ended up just adding it into the checks done by that test, but let me know if you'd rather split it into its own test case, totally understand if so.

**Comment by dpaasman00** [comment]
CI going to fail until I rebase once this is merged: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35591

**Review by evan-bradley — APPROVED** [review]

**Comment by evan-bradley** [comment]
This should be good to go after a rebase.

File: cmd/opampsupervisor/supervisor/commander/commander.go
Comment (evan-bradley) [reply cmd/opampsupervisor/supervisor/commander/commander.go:1]: That's okay with me, thanks.

**Pull request merged**
